osu.Game/Input/KeyBindingStore.cs

```diff
@@ -60,6 +63,18 @@ private void insertDefaults(IEnumerable<KeyBinding> defaults, int? rulesetId = n
                         usage.Context.SaveChanges();
                     }
                 }
+
+                // for now this should be fine here
+                foreach (var group in defaults.GroupBy(k => k.Action))
+                {
+                    foreach (var databasedKeyBinding in usage.Context.DatabasedKeyBinding.Where(b => b.RulesetID == rulesetId && b.Variant == variant && b.IntAction == (int)group.Key))
+                    {
+                        if (databasedKeyBinding.ActionName != null) continue;
+
+                        databasedKeyBinding.ActionName = group.Key.ToString();
```

Comment:
this is writing every time, regardless of current state. if this is supposed to be migrating, it should only target the required rows, in the `where` statement.

Reply:
Did not think about that. Will fix, thanks!